ionic_user_interface/src/components/UserRouteList.vue

```diff
@@ -0,0 +1,148 @@
+<template>
+  <ion-list class="ion-no-margin ion-no-padding">
+    <ion-card
+      v-for="(route, index) of routes"
+      :key="index"
+      class="ion-text-left route-card"
+      @click="() => handleRouteCardClick(route.username, route.createdAt)"
+    >
+      <ion-card-header>
+        <ion-card-title>{{ route.routeName }}</ion-card-title>
+        <VoteButton
+          class="vote-button"
+          :username="route.username"
+          :createdAt="route.createdAt"
+          v-model:voteCount="route.voteCount"
+          v-model:hasVoted="route.hasVoted"
+        ></VoteButton>
+      </ion-card-header>
+      <ion-card-content>
+        <b>Gym:&nbsp;</b>
+        <router-link
+          :to="'/gym/' + route.gymLocation + '/' + route.gymName"
+          @click.capture.stop="() => undefined"
+        >
+          {{ route.gymName }}
+        </router-link>
+        <br />
+        <b>Grade:</b>
+        V{{ route.publicGrade }}
+        <br />
+        <b>Created:</b>
+        {{ route.createdAt.split('T')[0] }}
+      </ion-card-content>
+    </ion-card>
+    <ion-card v-if="routes.length === 0" class="ion-text-center route-card">
+      <ion-card-header>
+        <ion-card-title>No Routes Found</ion-card-title>
+      </ion-card-header>
+    </ion-card>
+  </ion-list>
+</template>
+
+<script lang="ts">
+import { defineComponent, inject, Ref, ref, onMounted, computed, watch } from 'vue';
+import { heart, heartOutline } from 'ionicons/icons';
+import { IonCard, IonCardContent, IonCardHeader, IonCardTitle, IonList } from '@ionic/vue';
+import { useRouter, useRoute } from 'vue-router';
+import axios from 'axios';
+
+import VoteButton from '@/components/VoteButton.vue';
+import { throttle } from 'lodash';
+
+interface UserRoute {
+  commentCount: number;
+  createdAt: string;
+  gymLocation: string;
+  publicGrade: number;
+  routeName: string;
+  username: string;
+  voteCount: number;
+  hasVoted: boolean;
+  countryCode: string;
+  gymName: string;
+}
+
+export default defineComponent({
+  name: 'UserRouteList',
+  components: {
+    IonCard,
+    IonCardContent,
+    IonCardHeader,
+    IonCardTitle,
+    VoteButton,
+    IonList,
+  },
+  setup() {
+    const route = useRoute();
+    const router = useRouter();
+    const routes = ref<Array<UserRoute>>([]);
+    const getLoggedIn: () => Ref<boolean> = inject('getLoggedIn', () => ref(false));
+    const getAccessToken: () => Ref<string> = inject('getAccessToken', () => ref(''));
+    const profileUsername = computed(() => route.params.username as string);
+
+    const updateRoutes = throttle(() => {
```

Comment:
Would suggest resetting routes.value to [] before sending the request, so that the old routes won't be shown at all (a little confusing)